react-main/src/app/screens/Login/index.js

```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import { Field, reduxForm } from 'redux-form';
+
+const required = value => (value ? undefined : 'Field required.');
+const email = value => {
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  console.log(re.test(value));
+  return re.test(value);
+};
+const minLenght8 = value => (value.length >= 8 ? undefined : 'Field must be at least 8 characters long.');
+
+class Login extends Component {
+  render() {
+    const { onSubmit } = this.props;
+    return (
+      <form onSubmit={onSubmit}>
+        <div>
```

Comment:
You can extract divs to a custom Field component.
It should receive a name, label, component, type and validation array.